src/worker.js

```diff
@@ -109,46 +99,42 @@ class Worker {
   }
 
   /**
-   * Runs the worker. If the task for the job fails, then this method will retry
-   * the task (with an exponential backoff) as set by the environment.
-   *
-   * @returns {Promise} Promise that is resolved once the task succeeds or
-   *   fails.
+   * validate job against schema if passed
+   * @return {Promise}
+   * @rejects {WorkerStopError} when job does not match schema
    */
-  run (): Promise<void> {
-    this._incMonitor('ponos')
-    const timer = this._createTimer()
-    const log = this.log.child({
-      method: 'run',
-      queue: this.queue,
-      job: this.job
+  _validateJob (): Promise<void> {
+    return Promise.try(() => {
+      if (this.jobSchema) {
+        joi.assert(this.job, this.jobSchema)
+      }
     })
-
+    .catch((err) => {
```

Comment:
I think we should property `catch` this error just in case. `if (err.isJoi) {}`...